### PR TITLE
Add test coverage for delegated hash bins target download

### LINKED ISSUE #1639: RepositorySimulator: add support for delegated hash bins
**Description of issue or feature request**:

Currently `repository_tool.py` provides an easy way to distribute a large number of target files over multiple delegated roles (hashed bins).  A feature described in PEP458 that would be used in warehouse.  https://github.com/theupdateframework/python-tuf/blob/2206fc917e46ff584dcae75e640e1144c8b9f078/tuf/repository_tool.py#L2494

Having such functionality in `RepositorySimulator` that uses `metadata api` will be useful for testing the newer code.

### PR DESCRIPTION
**Pull request by ivanayov**
This change adds tests coverage for `path_hash_prefixes` and
verifies that role names matching specific prefixed successfully
find and download the corresponding metadata files and do not
succeed to find existing, but non-matching files

Fixes https://github.com/theupdateframework/python-tuf/issues/1639

Follow-up of #1808 

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [x] Tests have been added for the bug fix or new feature
- [n/a] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by jku** [comment]
I've not given this the review it needed, sorry... This vaguely looks like we could do better but I can't immediately see the better way: did you consider having the same test setup as in the existing downloads test `test_targetfile_search()` -- so there would be
* staticly defined metadata structure: it could be the same delegations_tree already defined in the file, just added with a targets metadata that does hashed bin delegations
* test cases would then only have to define a target path to search for, search result and the expected visited order

This would not make the current test much shorter but it might mean that adding a new test case would be maybe 2 lines instead of 40 lines of test data that it is now?

File: tests/test_updater_delegation_graphs.py [hunk 1]
```diff
@@ -438,6 +438,84 @@ def test_targetfile_search(self, test_data: TargetTestCase) -> None:
...
+                    b"fff0838676c1dacb8827b02511e89f85b43814a74dd2c3ffb31d952a1db94311",
+                    "non_matching_path_to_prefix.ext",
+                ),
+                TestTarget(
+                    "20-27",
+                    b"22992ae806e6d652835f3c84e169f05127da69e599f902e4dbdc6ff27ce46d01",
+                    "delegate_1.ext",
+                ),
+            ],
+            visited_order=["f8-ff", "20-27"],
```
Comment (MVrachev) [inline tests/test_updater_delegation_graphs.py:487]: I just noticed you are not using `visited_order` anywhere in your test.

**Comment by lukpueh** [comment]
Is this superseded by #2031? Should we think about deprecating `path_hash_prefixes` now that TAP 15 has landed? (see see https://github.com/theupdateframework/taps/issues/132#issuecomment-1115175226)

**Comment by jku** [comment]
> Is this superseded by #2031? 

I think so. I wouldn't object to a tight client test specifically for hashed bins but I don't think it's super important, and this PR is 80 lines of test code for very limited (new) test functionality.

**Pull request closed without merging**